Adds basic documentation for the ToolbarButton component.

mateuswetah opened the pull request:
## Description
Adds basic documentation for the `ToolbarButton` component. Any review and suggestion is welcome @mkaz, this is my first trial here.

mateuswetah committed:
Adds basic documentation for the ToolbarButton component.

mateuswetah committed:
Updates compiled version of docs with ToolbarButton docs.

talldan commented on packages/components/src/toolbar-button/README.md:
```diff
@@ -0,0 +1,26 @@
+# ToolbarButton
+
+A ToolbarButton can be used to add actions to your block control, usually inside a ToolbarGroup. It has the same features of the [Button](https://developer.wordpress.org/block-editor/components/button/) component, only its style is more appropriate for a Toolbar.
```
> only its style is more appropriate for a Toolbar.

Hey @mateuswetah, thanks for working on this! 

I think there are a few extra features of the `ToolbarButton` over just styling. It also has some additional accessibility functionality implemented via the Reakit library - https://reakit.io/docs/toolbar/. As mentioned there, it implements this toolbar pattern for keyboard interactions - https://www.w3.org/TR/wai-aria-practices/#toolbar.

```suggestion
A ToolbarButton can be used to add actions to your block control, usually inside a ToolbarGroup. It has similar features to the [Button](https://developer.wordpress.org/block-editor/components/button/) component. Using `ToolbarButton` will ensure the correct styling for a button in a toolbar, and also that keyboard interactions in a toolabr are consistent with the [WAI ARIA toolbar pattern](https://www.w3.org/TR/wai-aria-practices/#toolbar).
```

mateuswetah replied:
Awesome, thanks for the clarification, I think its better :)

talldan commented on packages/components/src/toolbar-button/README.md:
```diff
@@ -0,0 +1,26 @@
+
+## Related components
+
+* If you wish to implement a control do select options grouped as icon buttons you can use the [Toolbar](https://developer.wordpress.org/block-editor/components/toolbar/) component, which already handles this strategy.
```
> control do select

Looks like a typo here.

mateuswetah replied:
Yeap, sorry for that, I'll pay more attention in future contributions!

mateuswetah committed:
Update packages/components/src/toolbar-button/README.md

Elaborates more on ToolbarComponent information, mentioning its accessibility features.

Co-authored-by: Daniel Richards <daniel.p.richards@gmail.com>

mateuswetah committed:
Fixes typo 'control do select' on ToolbarButton documentation.

mkaz commented on packages/components/src/toolbar-button/README.md:
```diff
@@ -0,0 +1,26 @@
+# ToolbarButton
+
+A ToolbarButton can be used to add actions to your block control, usually inside a ToolbarGroup. It has similar features to the [Button](https://developer.wordpress.org/block-editor/components/button/) component. Using `ToolbarButton` will ensure the correct styling for a button in a toolbar, and also that keyboard interactions in a toolbar are consistent with the [WAI ARIA toolbar pattern](https ...
```
For linking to components, you want to not include the full domain, since the docs are browsed in multiple contexts. See [the Using Links section](https://developer.wordpress.org/block-editor/contributors/document/#using-links) of the contributing docs.

```suggestion
A ToolbarButton can be used to add actions to your block control, usually inside a ToolbarGroup. It has similar features to the [Button](/packages/components/src/button/README.md) component. Using `ToolbarButton` will ensure the correct styling for a button in a toolbar, and also that keyboard interactions in a toolbar are consistent with the [WAI ARIA toolbar pattern](https://www.w3.org/TR/wai-aria-practices/#toolbar).
```

mateuswetah replied:
Thanks @mkaz, I haven't read that page, will check it out. I'm committing all this suggestions!

mkaz commented on packages/components/src/toolbar-button/README.md:
```diff
@@ -0,0 +1,26 @@
+
+## Props
+
+This component accepts [the same API of the Button](https://developer.wordpress.org/block-editor/components/button/#props) component.
```
For linking to components, you want to not include the full domain, since the docs are browsed in multiple contexts. See [the Using Links section](https://developer.wordpress.org/block-editor/contributors/document/#using-links) of the contributing docs.

```suggestion
This component accepts [the same API of the Button](/packages/components/src/button/README.md#props) component.
```

mkaz commented on packages/components/src/toolbar-button/README.md:
```diff
@@ -0,0 +1,26 @@
+
+## Related components
+
+* If you wish to implement a control to select options grouped as icon buttons you can use the [Toolbar](https://developer.wordpress.org/block-editor/components/toolbar/) component, which already handles this strategy.
```
Same as above.

```suggestion
* If you wish to implement a control to select options grouped as icon buttons you can use the [Toolbar](/packages/components/src/toolbar/README.md) component, which already handles this strategy.
```

mkaz commented on packages/components/src/toolbar-button/README.md:
```diff
@@ -0,0 +1,26 @@
+## Related components
+
+* If you wish to implement a control to select options grouped as icon buttons you can use the [Toolbar](https://developer.wordpress.org/block-editor/components/toolbar/) component, which already handles this strategy.
+* The ToolbarButton may be used with other elements such as [Dropdown](https://developer.wordpress.org/block-editor/components/dropdown/) to display options in a popover.
```
Same as above.

```suggestion
* The ToolbarButton may be used with other elements such as [Dropdown](/packages/components/src/dropdown/README.md) to display options in a popover.
```

mateuswetah committed:
Update packages/components/src/toolbar-button/README.md

Proper documentation links on ToolbarButton page.

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

mateuswetah committed:
Update packages/components/src/toolbar-button/README.md

Proper links usage on ToolbarButton documentation page.

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

mateuswetah committed:
Update packages/components/src/toolbar-button/README.md

Proper links usage on ToolbarButton documentation page.

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

mateuswetah committed:
Update packages/components/src/toolbar-button/README.md

Proper links usage on ToolbarButton documentation page.

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

diegohaz reviewed (approved):
Nice! Thanks for working on this, Mateus! Documentation improvements are always welcome! ❤️ 🇧🇷
